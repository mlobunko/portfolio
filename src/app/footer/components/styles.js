import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const StyledFontAwesomeIcon = styled(FontAwesomeIcon)`
  color: rgb(85, 85, 85);
`;
StyledFontAwesomeIcon.displayName = "StyledFontAwesomeIcon";

export const StyledFooter = styled.footer`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 6rem;
`;
StyledFooter.displayName = "StyledFooter";

export const StyledSocial = styled.div`
  ${props => `url(${props.image}) no-repeat center center / contain`};
  width: 4rem;
  height: 100%;
  margin: 0 1rem;
  font-size: 2rem;
  text-align: center;
`;
StyledSocial.displayName = "StyledSocial";
