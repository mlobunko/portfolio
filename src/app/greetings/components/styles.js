import styled from "styled-components";

export const StyledArrow = styled.div`
  margin-top: 0;
  color: rgb(100, 100, 100);
`;
StyledArrow.displayName = "StyledArrow";

export const StyledBottom = styled.div`
  text-align: center;
  font-size: 0.9rem;
  font-weight: 100;
  color: black;
  padding-bottom: 10vh;

  @media (min-width: 375px) {
    font-size: 1rem;
  }}
`;
StyledBottom.displayName = "StyledBottom";

export const StyledBottomText = styled.div`
  margin-bottom: 0.3rem;
`;
StyledBottomText.displayName = "StyledBottomText";

export const StyledGreetings = styled.div`
  background: ${props => `url(${props.image}) no-repeat left center / cover`};
  display: flex;
  flex-direction: column;
  height: 100vh;
  align-content: center;
`;
StyledGreetings.displayName = "StyledGreetings";

export const StyledHeader = styled.div`
  text-align: center;
  font-size: 2.5rem;
  font-weight: 100;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 1rem;

  @media (min-width: 768px) {
    font-size: 3rem;
  }

  @media (min-width: 1024px) {
    font-size: 4rem;
  }
`;
StyledHeader.displayName = "StyledHeader";

export const StyledHeaderFirstSentence = styled.h1`
  padding-bottom: 2rem;
  color: inherit;
  font-size: inherit;
  font-weight: inherit;
  margin: 0;
`;
StyledHeaderFirstSentence.displayName = "StyledHeaderFirstSentence";

export const StyledHeaderSecondSentence = styled.h2`
  color: inherit;
  font-size: inherit;
  font-weight: inherit;
  margin: 0;
`;
StyledHeaderSecondSentence.displayName = "StyledHeaderSecondSentence";
