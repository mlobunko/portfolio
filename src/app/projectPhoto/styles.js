import styled from "styled-components";

export const StyledDescription = styled.div`
  text-align: center;
  font-weight: 400;
  padding-bottom: 1rem;
`;
StyledDescription.displayName = "StyledDescription";

export const StyledNameOfProject = styled.h2`
  text-align: center;
  font-weight: 500;
  font-size: 2rem;
  padding-bottom: 1rem;
  margin: 0;

  @media (min-width: 768px) {
    font-size: 3rem;
  }
`;
StyledNameOfProject.displayName = "StyledNameOfProject";

export const StyledProjectPhoto = styled.section`
  width: 100%;
  height: 600px;
  max-width: 1366px;
  color: ${props => (props.textColor ? `${props.textColor}` : "black")};
  margin-bottom: ${props => props.margin && "2rem"};
  margin-left: auto;
  margin-right: auto;
  background: ${props => `url(${props.image}) no-repeat center center / cover`};
  display: flex;
  flex-direction: column;
  justify-content: ${props => (props.textUp ? "flex-start" : "flex-end")};

  @media (min-width: 768px) {
    justify-content: flex-start;
  }
`;
StyledProjectPhoto.displayName = "StyledProjectPhoto";

export const StyledText = styled.div`
  margin: 0 auto;
  width: 18rem;
  padding-bottom: ${props => (props.textUp ? "0" : "1rem")};
  padding-top: ${props => (props.textUp ? "1rem" : "0")};
  text-align: center;
  @media (min-width: 375px) {
    width: 20rem;
  }
  @media (min-width: 768px) {
    margin-left: 5%;
    padding-top: 2rem;
    padding-bottom: 0;
  }
`;
StyledText.displayName = "StyledText";
