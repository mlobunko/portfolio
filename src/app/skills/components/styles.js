import styled from "styled-components";

export const StyledImage = styled.div`
  background: ${props =>
    `url(${props.image}) no-repeat center center / contain`};
  width: 100%;
  height: 3rem;
  margin-bottom: 0.2rem;

  @media (min-width: 768px) {
    height: 6rem;
  }
`;
StyledImage.displayName = "StyledImage";

export const StyledSkill = styled.div`
  height: 4rem;
  width: 40%;
  text-align: center;

  @media (min-width: 768px) {
    width: 25%;
    height: 7rem;
  }
`;
StyledSkill.displayName = "StyledSkill";

export const StyledSkills = styled.div`
  width: 100%;
  height: 600px;
  max-width: 1366px;
  margin: 0 auto;
  background: white;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-content: space-around;
`;
StyledSkills.displayName = "Skills";
