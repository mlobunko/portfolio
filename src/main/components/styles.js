import styled, { injectGlobal } from "styled-components";
import RobotoMedium from "../fonts/Roboto-Medium.ttf";
import RobotoRegular from "../fonts/Roboto-Regular.ttf";
import RobotoThin from "../fonts/Roboto-Thin.ttf";

injectGlobal`
  *, *:before, *:after {
    box-sizing: border-box;
  }
`;

export const StyledHeaderOfSections = styled.h2`
  padding: 2rem 0;
  margin: 0;
  font-size: 1.5rem;
  font-weight: 100;
  text-align: center;
  letter-spacing: 0.15rem;

  @media (min-width: 768px) {
    font-size: 2rem;
  }
`;
StyledHeaderOfSections.displayName = "StyledHeaderOfSections";

export const StyledMain = styled.div`
  background: rgb(245, 245, 245);
  color: black;
  font-family: "Roboto", Helvetica, Arial, sans-serif;
  font-weight: 400;
  min-height: 100vh;

  @font-face {
    font-family: Roboto;
    src: url(${RobotoThin});
    font-weight: 100;
  }

  @font-face {
    font-family: Roboto;
    src: url(${RobotoRegular});
    font-weight: 400;
  }

  @font-face {
    font-family: Roboto;
    src: url(${RobotoMedium});
    font-weight: 500;
  }
`;
StyledMain.displayName = "StyledMain";
