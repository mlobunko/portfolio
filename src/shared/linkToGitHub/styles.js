import styled from "styled-components";

export const StyledLink = styled.a.attrs({
  href: props => props.href,
  target: "_blank",
  rel: "noopener noreferrer",
  "aria-label": props => `Link to ${props.nameOfProject}`
})`
  color: ${props =>
    props.textColor ? `${props.textColor}` : "rgb(0, 112, 201)"};
  text-decoration: none;
  font-weight: 500;

  &:hover {
    color: ${props =>
      props.textColor ? `${props.textColor}` : "rgb(0, 112, 201)"};
    text-decoration: underline;
    cursor: pointer;
  }
  &:active {
    color: ${props =>
      props.textColor ? `${props.textColor}` : "rgb(0, 112, 201)"};
    text-decoration: underline;
    cursor: pointer;
  }
  &:visited {
    color: ${props =>
      props.textColor ? `${props.textColor}` : "rgb(0, 112, 201)"};
    text-decoration: none;
    cursor: auto;
  }
  &::after {
    display: inline-block;
    content: ">";
    margin-left: 0.3rem;
  }
`;

StyledLink.displayName = "StyledLink";
