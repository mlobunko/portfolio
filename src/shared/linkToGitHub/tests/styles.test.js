import React from "react";
import { shallow } from "enzyme";
import "enzyme-to-json";

import { StyledLink } from "../styles";

describe("StyledLink", () => {
  const initialProps = {
    href: "123",
    nameOfProject: "family",
    textColor: "black"
  };
  it("renders correctly, all properties exists", () => {
    const props = {
      ...initialProps
    };
    const wrapper = shallow(<StyledLink {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly, textColor false", () => {
    const props = {
      ...initialProps,
      textColor: ""
    };
    const wrapper = shallow(<StyledLink {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
