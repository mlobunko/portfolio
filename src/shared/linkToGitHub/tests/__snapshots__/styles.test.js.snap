// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLink renders correctly, all properties exists 1`] = `
<a
  aria-label="Link to family"
  className="sc-bdVaJa iOrTop"
  href="123"
  rel="noopener noreferrer"
  target="_blank"
/>
`;

exports[`StyledLink renders correctly, textColor false 1`] = `
<a
  aria-label="Link to family"
  className="sc-bdVaJa efIPLF"
  href="123"
  rel="noopener noreferrer"
  target="_blank"
/>
`;
